cdap-metadata-spi/src/main/java/io/cdap/cdap/spi/metadata/MetadataConsumer.java

```diff
@@ -24,9 +24,21 @@
 @Beta
 public interface MetadataConsumer {
 
+  /**
+   * @return the name of the metadata consumer.
+   */
+  String getName();
+
   /**
    * Consumes lineage info for a program run.
+   *
+   * @param context context for the MetadataConsumer containing properties required by the implementation.
+   * @param programRun the {@link ProgramRun} for which lineage is consumed
+   * @param lineageInfo the {@link LineageInfo} containing details of lineage for the program run
+   *
+   * @throws IllegalArgumentException if the context is missing any required properties
    */
-  void consumeLineage(ProgramRun programRun, LineageInfo lineageInfo);
+  void consumeLineage(MetadataConsumerContext context, ProgramRun programRun, LineageInfo lineageInfo)
+    throws Exception;
```

Comment:
If method only throws unchecked `IllegalArgumentException` you don't need to add `throws Exception` to method signature. This will cause clients to unnecessarily call this method in try..catch block always

Reply:
Changed as discussed offline